From 166cd2ad62ca0115d62009f9c590f1bb95ed5ca3 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Fri, 23 Dec 2016 09:48:46 -0800
Subject: [PATCH] joyent/node-cueball#65 CueBallAgent constructor allows
 optional log object but calls it as if it were mandatory

---
 lib/agent.js    |  8 +++++++-
 lib/pool.js     |  3 ++-
 lib/resolver.js | 12 +++++++++---
 lib/set.js      |  3 ++-
 package.json    |  2 +-
 5 files changed, 21 insertions(+), 7 deletions(-)

diff --git a/lib/agent.js b/lib/agent.js
index a8a581b..a8fafd8 100644
--- a/lib/agent.js
+++ b/lib/agent.js
@@ -22,6 +22,7 @@ const mod_http = require('http');
 const mod_https = require('https');
 const mod_utils = require('./utils');
 const mod_vasync = require('vasync');
+const mod_bunyan = require('bunyan');
 
 const Pool = mod_pool.ConnectionPool;
 
@@ -52,7 +53,12 @@ function CueBallAgent(options) {
 
 	this.pools = {};
 	this.resolvers = options.resolvers;
-	this.log = options.log;
+	this.log = options.log || mod_bunyan.createLogger({
+		name: 'cueball'
+	});
+	this.log = this.log.child({
+		component: 'CueBallAgent'
+	});
 	this.cba_stopped = false;
 
 	mod_assert.number(options.spares, 'options.spares');
diff --git a/lib/pool.js b/lib/pool.js
index 1ec3503..e4120bf 100644
--- a/lib/pool.js
+++ b/lib/pool.js
@@ -142,9 +142,10 @@ function CueBallConnectionPool(options) {
 
 	mod_assert.optionalObject(options.log, 'options.log');
 	this.p_log = options.log || mod_bunyan.createLogger({
-		name: 'CueBallConnectionPool'
+		name: 'cueball'
 	});
 	this.p_log = this.p_log.child({
+		component: 'CueBallConnectionPool',
 		domain: options.domain,
 		service: options.service,
 		pool: this.p_uuid
diff --git a/lib/resolver.js b/lib/resolver.js
index 840d67a..666921e 100644
--- a/lib/resolver.js
+++ b/lib/resolver.js
@@ -72,7 +72,10 @@ function CueBallResolver(fsm, options) {
 
 	mod_assert.optionalObject(options.log, 'options.log');
 	this.r_log = options.log || mod_bunyan.createLogger({
-		name: 'CueBallResolver'
+		name: 'cueball'
+	});
+	this.r_log = this.r_log.child({
+		component: 'CueBallResolver'
 	});
 
 	FSM.call(this, 'stopped');
@@ -263,9 +266,12 @@ function CueBallDNSResolver(options) {
 
 	mod_assert.optionalObject(options.log, 'options.log');
 	this.r_log = options.log || mod_bunyan.createLogger({
-		name: 'CueBallDNSResolver'
+		name: 'cueball'
+	});
+	this.r_log = this.r_log.child({
+		component: 'CueBallDNSResolver',
+		domain: this.r_domain
 	});
-	this.r_log = this.r_log.child({domain: this.r_domain});
 
 	mod_assert.object(options.recovery, 'options.recovery');
 	this.r_recovery = options.recovery;
diff --git a/lib/set.js b/lib/set.js
index 10d7c89..03e2c0c 100644
--- a/lib/set.js
+++ b/lib/set.js
@@ -46,9 +46,10 @@ function CueBallConnectionSet(options) {
 
 	mod_assert.optionalObject(options.log, 'options.log');
 	this.cs_log = options.log || mod_bunyan.createLogger({
-		name: 'CueBallConnectionPool'
+		name: 'cueball'
 	});
 	this.cs_log = this.cs_log.child({
+		component: 'CueBallConnectionSet',
 		domain: options.domain,
 		service: options.service,
 		cset: this.cs_uuid
diff --git a/package.json b/package.json
index 0df5f04..ca476eb 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "1.2.3",
+  "version": "1.2.4",
   "description": "",
   "main": "lib/index.js",
   "dependencies": {
-- 
2.21.0

